chalk-ir/src/debug.rs

```diff
@@ -23,78 +23,63 @@ impl<I: Interner> Debug for AssocTypeId<I> {
 
 impl<I: Interner> Debug for Ty<I> {
     fn fmt(&self, fmt: &mut Formatter<'_>) -> Result<(), Error> {
-        I::debug_ty(self, fmt)
-            .unwrap_or_else(|| unimplemented!("cannot format Ty without setting Program in tls"))
+        I::debug_ty(self, fmt).unwrap_or_else(|| write!(fmt, "Ty(?)"))
```

Comment:
```suggestion
        I::debug_ty(self, fmt).unwrap_or_else(|| write!(fmt, "{:?}", self.interned))
```

Weren't we going to print the 'interned' format instead? Something like this?

Reply:
🤦‍♀️I forgot those are `: Debug`